### ISSUE TITLE
Rebuilding the Docs 

### ISSUE THREAD

**Issue by livghit** [a-h/templ#153]
Hi 🌊  🐙 !
Hope you are doing well ! First I wanted to thank you for this cool powerful Templating language !! 

I think it would be very cool if we would be transferring the docs to use Templ itself to show the power of it :) ! I would be ready to help with that ! 

Have a nice weekend !

**Comment by livghit** [comment]
With this , I would also try and create an easy to implement  Components library where people could plug into the project and use it .

**Comment by joerdav** [comment]
This is actually something we've thought about. A documentation engine powered by templ would be a great way to dog-food our own project!

It was on the roadmap once a few bigger issues are ironed out, but don't let that stop you! Happy to support any code being put towards the templ docs.

I think templ is a gread tool to support a component library, you can distribute it as a go mod, and import it as you would any other go dependency.

**Comment by headblockhead** [comment]
This sounds like a great idea, perhaps it could allow for the command-line output/generated files of `templ` commands to be autofilled into the docs instead of manual copy-pasting.

**Comment by PentesterPriyanshu** [comment]
can anyone still working on it

**Comment by livghit** [comment]
At the moment I do not really have the resources for this . Altho I do not want to give it up because I really enjoy the project. I will try to make me some more time and start more intensively on it 🐙

**Comment by cugu** [comment]
I played with templ a bit today and built a [simple static site generator](https://github.com/cugu/templum) and rendered the [templ docs](https://cugu.github.io/templum-example/).

Happy to fix the remaining bugs and integrate it if you are interested.

Anyway templ is a great tool, thanks for it!

**Comment by a-h** [comment]
Is this completed? Doesn't feel like it, since templ still uses Docusaurus. :)

**Comment by livghit** [comment]
Sorry ... I did not close this issue on purpose ....

**Comment by livghit** [comment]
@a-h  how about @cugu implementation it look great to me ?

**Comment by joerdav** [comment]
It does look very good, however I think at the moment there may be some rough edges still that @cugu is ironing out. There is a build for the templ docs using templum that shows a couple of them:

https://cugu.github.io/templum-example/

**Comment by cugu** [comment]
I moved to another syntax highlighter as there was not much drive for adopting templum here recently. I changed that back, so that https://cugu.github.io/templum-example/index.html should now be correct (just outdated).

If there is further interest in using templum I am happy to iron out issues, but I don't plan include as many features as Docusaurus, hugo or similar. It should stay a rather simple templ project. We could also move it to https://github.com/templ-go, if you like.

**Comment by joerdav** [comment]
I think the only step left is to migrate now. The syntax highlighting was the only issue really that I saw

**Comment by cugu** [comment]
Great, I'll create a PR.

Would you prefer to move it to https://github.com/templ-go, include the whole project here, or keep it where is?

**Comment by a-h** [comment]
What's the plan? Create a general static documentation generation project that templ uses, or create the templ docs?

If it's the first, then I'd say make the docs generator on your own GitHub user name, because I think that templ's job is to be a good way to make HTML, and it's out of scope for templ to equal Hugo or Docusaurus.

Templ can promote it, and the docs site can link to it, but you'd have a maintenance job on your hands if it gets popular. 😀 

If it's the 2nd, then docs live within the code, so the two are together. In which case a PR is great. Id like to run side by side for a while to reduce risk. For example, you see the new site, but there's a choice to use the old Docusaurus version so that issues with the new one don't stop anyone from finding the info they need.

I'm not sure how we'd do this at the moment. The docs are currently GitHub pages, but I can redirect the domain to a Fargate container or similar if we need to use CPU for that.

**Comment by cugu** [comment]
I opted for plan 2 and started https://github.com/a-h/templ/pull/401.

You can see the generated docs at: https://cugu.github.io/templ/new/

The docusaurus can live in parallel on the same GitHub pages, as long as no directory or page is named `new` in the docs.
